pkg/services/govmomi/util.go

```diff
@@ -457,6 +457,11 @@ func waitForIPAddresses(
 				// Look at each IP and determine whether a reconcile has
 				// been triggered for the IP.
 				for _, discoveredIPInfo := range nic.IpConfig.IpAddress {
+					// Ignore device if SkipIPAllocation is set.
+					if deviceSpec.SkipIPAllocation {
```

Comment:
Not sure if we want to run continue in this first case. Wouldn't this also skip discovering if IP Addresses get set on this interface?

If so, we should add it to the description that IPs for this device don't get bubbled up.

Reply:
I think you're right. Even if there is IP address discovered but the device has no static IP or DHCP configured, the discovered address will be skipped anyway.
Deleted this if statement.